### ISSUE TITLE
Add support for symbols as keys for $merge

### ISSUE THREAD

**Issue by nponiros** [kolodny/immutability-helper#9]
Hi there,

as it seems the `$merge` command currently does not support symbols for the keys. Probably because of [line 96](https://github.com/kolodny/immutability-helper/blob/master/index.js#L96). Adding support for Symbol should be easy, at least if only modern browsers are considered with native Symbol support. For older browsers it will be trickier and polyfills must be used.

If you are interested in this, I would gladly create a pull request and we can see whether it is good enough and how we can improve it.

**Comment by kolodny** [comment]
This seems like a edge case for this. I'd recommend doing something like this:

```js
var myUpdate = require('./').newContext();

var obj = {
  [Symbol.for('s')]: 'sym',
  key: 'value'
}
var getAllKeys = obj => Object.getOwnPropertySymbols(obj).concat(Object.keys(obj))
myUpdate.extend('$merge', function(value, original) {
  var newObject = Object.assign({}, original);
  getAllKeys (value).forEach(key => newObject[key] = value[key]);
  return newObject
})
var next = myUpdate(obj, {$merge: { c: 'test' }})
console.log({
  "[Symbol.for('s')]": next[Symbol.for('s')],
  key: next['key'],
  c: next['c'],
})
// { '[Symbol.for(\'s\')]': 'sym', key: 'value', c: 'test' }
```

**Comment by nponiros** [comment]
It might be an edge case now but I would assume that in the future more people will use Symbols so it might not be an edge case then.

In any case, I use `$merge` to avoid multiple `Object.assign` calls and that does support Symbols so I expected `$merge` to do the same. If you don't want to support Symbols then maybe the workaround you posted should be documented. I saw that you started documenting the library in an other branch. Maybe I can help you with that as I think that the library can benefit from a more thorough documentation. Currently it is not really usable if the user does not read the react counterpart, which imho is also not very well documented. Some more examples would be nice there.

Anyway if you need any help, just let me know.

**Comment by kolodny** [comment]
I just pushed a branch https://github.com/kolodny/immutability-helper/tree/symbols to add support for them. Let me know if that covers what you had it mind.

**Comment by nponiros** [comment]
Looks good to me ☺

**Comment by kolodny** [comment]
Published in `2.1.0`
Thanks!
